src/ess/reflectometry/corrections.py

```diff
@@ -29,91 +30,63 @@ def footprint_correction(
-    return IofQ[Sample](normalize_by_counts(data_array))
-
-
-def normalize_reference(
-    data_array: HistogrammedQData[Reference],
-    calibration_factor: SupermirrorCalibrationFactor,
-) -> IofQ[Reference]:
-    return IofQ[Reference](normalize_by_counts(calibration_factor * data_array))
-
-
-def normalize_by_counts(
-    data_array: sc.DataArray,
-) -> sc.DataArray:
-    """
-    Normalize the bin-summed data by the total number of counts.
-    If the data has variances, a check is performed to ensure that the counts in each
-    bin is much lower than the total counts. If this is not the case, an error is raised
-    because the normalization would introduce non-negligible correlations which are not
-    handled Scipp's basic error propagation. See Heybrock et al. (2023).
-    If the check passes, the input data is simply divided by the total number of counts,
-    ignoring the variances of the denominator.
-
-    Parameters
-    ----------
-    data_array:
-        Data array to be normalized.
-
-    Returns
-    -------
-    :
-        Normalized data array.
+def compute_reference_intensity(
+    da: FootprintCorrectedData[Reference], wb: WBins
+) -> HistogrammedReference:
+    """Creates a reference intensity map over (z_index, wavelength).
+    Rationale:
+        The intensity expressed in those variables should not vary
+        with the experiment parameters (such as sample rotation).
+        Therefore it can be used to normalize sample measurements.
     """
-    # Dividing by ncounts fails because ncounts also has variances, and this introduces
-    # correlations. According to Heybrock et al. (2023), we can however safely drop the
-    # variances of ncounts if counts_in_bin / ncounts is small everywhere.
-    ncounts = sc.values(data_array.sum())
-    norm = data_array / ncounts
-    if (data_array.variances is not None) and (norm.max().value > 0.1):
-        ind = np.argmax(data_array.values)
-        raise ValueError(
-            'One or more bins contain a number of counts of the same order as the '
-            'total number of counts. It is not safe to drop the variances of the '
-            'denominator when normalizing by the total number of counts in this '
-            f'regime. The maximum counts found is {data_array.values[ind]} at '
-            f'index {ind}. The total number of counts is {ncounts.value}.'
+    # TODO: use lower level function to histogram without binning
+    h = da.group('z_index').hist(wavelength=wb)
+    h.masks['too_few_events'] = h.data < sc.scalar(1, unit='counts')
+    h.coords['wavelength'] = sc.midpoints(h.coords['wavelength'])
+    # Add a Q coordinate to each bin, the Q is not completely unique in every bin,
+    # but it is close enough.
+    h.coords['Q'] = (
+        sc.DataArray(
+            data=da.coords['Q'],
+            coords=dict(
+                wavelength=da.coords['wavelength'], z_index=da.coords['z_index']
+            ),
+            masks=da.masks,
```

Comment:
Is this for method 2.) you describe in a message earlier? If so, I presume this is going to change?

Reply:
It's actually needed for both methods.
It's needed to compute
$$
    \int_{Q(z, \lambda) \in Q_i} \ldots \ dz \ d\lambda
$$

Basically we need to compute $Q(z, \lambda)$ but the way $Q$ is computed it is not a function of $z$ and $\lambda$ because it also depends on $y$ (and $x$?). However, to good approximation it is a function of $z$ and $\lambda$.

The alternative to the current implementation is to define an explicit expression for $Q(z, \lambda)$.

Reply:
Then maybe we should compute Q correctly, independent of `y`? Since this is not GISANS?

Reply:
What do you mean by "correctly"? $Q$ depends on $\lambda$, $\lambda$ depends on $L2$, $L2$ depends on the pixel position, and the pixel position depends on $y$.

Reply:
If we think of the incoming beam as a plane wave (as I suppose it is assumed to be, at least for the illuminated area?) then one can define things in such a way that it is essentially doing 2-D math, instead of 3-D.

Reply:
Sure we could approximate `Q` in another way so that $Q(z, \lambda)$ is uniquely defined, but I don't really see the point? What's the issue we're solving?

Reply:
My thinking was that this would *avoid* the need for approximation?

Reply:
Assuming the current implementation of $Q$ is not just incorrect, you would just be replacing one approximation in this step with another approximation in an earlier step.

Maybe that's a compromise that strikes a better balance between correctness, performance, and simplicity. In that case I think we should absolutely make that change.